src/compiler/commandLineParser.ts

```diff
@@ -1704,7 +1697,7 @@ namespace ts {
             errors.push(createDiagnostic(Diagnostics.A_path_in_an_extends_option_must_be_relative_or_rooted_but_0_is_not, extendedConfig));
             return undefined;
         }
-        let extendedConfigPath = toPath(extendedConfig, basePath, getCanonicalFileName);
+        let extendedConfigPath = toPath(extendedConfig, basePath, identity);

```

Comment:
I dont this is right.. Especially on case sensitive file system. You would want to make sure ```Path``` has correct casing depending on file system casing. The bug i think is we are are using Path in place of NormalizedPath(where the path is not in canonical form - handled casing) when surfacing error or reading from file system

Reply:
What we want to do is preserve all input casing we have been given until we actually try to reify the file names into source files after the config file(s) have been fully dealt with. That means explicitly _not_ canonicalizing (aka unifying disparate cases on case-insensitive systems) any path at any point during config resolution. We _must_ do so in order to maintain the casing of the path fragments we're working with as we traverse the filesystem while finding and parsing config files before we create `forceConsistentCasingInFileNames` errors.

Reply:
I do understand that but ```Path``` is a type that is used to represent canonicalized paths and that's not correct here. You could use ```getNormalizedAbsolutePath``` to avoid case conversions...